bundles/org.openhab.voice.pipertts/src/main/java/org/openhab/voice/pipertts/internal/PiperTTSService.java

```diff
@@ -93,17 +113,89 @@ public PiperTTSService(final @Reference TTSCache ttsCache) {
 
     @Activate
     protected void activate(Map<String, Object> config) {
-        try {
-            piper = new PiperJNI();
-            piper.initialize(true, false);
-            logger.debug("Using Piper version {}", piper.getPiperVersion());
-        } catch (IOException e) {
-            logger.warn("Piper registration failed, the add-on will not work: {}", e.getMessage());
-        }
+        activateTask = executor.submit(() -> {
+            try {
+                setupNativeDependencies();
+                piper = new PiperJNI();
+                piper.initialize(true, false);
+                logger.debug("Using Piper version {}", piper.getPiperVersion());
+                ready = true;
+            } catch (IOException e) {
+                logger.warn("Piper registration failed, the add-on will not work: {}", e.getMessage());
+            }
+        });
         tryCreatePiperDirectory();
         configChange(config);
     }
 
+    @Deactivate
+    private void deactivate() {
+        if (activateTask != null && !activateTask.isDone()) {
+            activateTask.cancel(true);
+        }
+    }
+
+    private void setupNativeDependencies() throws IOException {
+        if (!Files.exists(LIB_FOLDER)) {
+            Files.createDirectory(LIB_FOLDER);
+        }
+        if (!Files.exists(JAR_FILE)) {
+            logger.debug("Downloading file: {}", JAR_URL);
+            InputStream in = new URL(JAR_URL).openStream();
+            Files.copy(in, JAR_FILE, StandardCopyOption.REPLACE_EXISTING);
+        }
```

Comment:
You could move this after checking the compatibility of the server. It will avoid downloading the jar file in case the platform is not compatible.